quic/s2n-quic-core/src/recovery/hybrid_slow_start.rs

```diff
@@ -94,24 +137,56 @@ impl HybridSlowStart {
         if let (N_SAMPLING, Some(last_min_rtt), Some(cur_min_rtt)) =
             (self.sample_count, self.last_min_rtt, self.cur_min_rtt)
         {
-            let threshold = last_min_rtt / THRESHOLD_DIVIDEND;
-            // Clamp n to the min and max thresholds
-            let threshold = threshold.min(MAX_DELAY_THRESHOLD).max(MIN_DELAY_THRESHOLD);
-            let delay_increase_is_over_threshold = cur_min_rtt >= last_min_rtt + threshold;
-            let congestion_window_is_above_minimum = congestion_window >= self.low_ssthresh();
-
-            if delay_increase_is_over_threshold && congestion_window_is_above_minimum {
-                self.threshold = congestion_window;
+            if congestion_window >= self.css_threshold {
+                self.css_count += 1;
+                if cur_min_rtt < self.css_baseline_min_rtt {
+                    // resume slow start
+                    self.css_threshold = self.threshold;
+                    self.ss_growth_divisor = 1.0;
+                    self.css_count = 0;
+                }
+                if self.css_count >= CSS_ROUNDS {
+                    // exit slow start phase
+                    self.threshold = congestion_window;
+                    self.css_threshold = f32::MAX;
+                    self.ss_growth_divisor = 1.0;
```

Comment:
wondering if you should just set `self.use_hystart_plus_plus` to false here given the above citation `An implementation SHOULD use HyStart++ only for the initial slow start`

Reply:
yeah you are right. Assuming hystart++ is shown to be performant and we end up removing regular hystart this should be fine, but if we end up wanting to keep both for some reason we might want to revisit that.

Reply:
Yep, I agree with you.